### ISSUE TITLE
No support running on Hyper-V (Not Virtual Box)  ? 

### ISSUE THREAD

**Issue by sg1970** [redhatdemocentral/ocp-install-demo#29]
just realized that this demo needs virtual box .... I know the older versions of this demo worked on hyper-v .. has that been discontinued ?  Any reasons for that ? Thanks.

**Comment by eschabell** [comment]
This is parameterized, it is default for osX and defaults to KVM for Linux but you can change if you like to something else supported, see [init.sh line 9 for default and line 55 for default Linux setting](https://github.com/redhatdemocentral/ocp-install-demo/blob/master/init.sh).

Does this help?

**Comment by sg1970** [comment]
@eschabell Thanks for taking the time to respond !

Unfortunately Iam **NOT** on Linux. Iam on a PC running Windows 10 Pro(ver=1607 , OS Build=14393.1593) . Using Hyper-V for virtualization.. But Docker is running in Linux mode ( using moby linux vm )  and behind a corporate proxy.


Docker Version = Version 17.06.1-ce-**win24** (13025)
Channel: stable
54dc09c

OC.exe client version = 3.6.173.0.5  ( Official version )

Command used in the **current** init.bat file:

`docker-machine create --driver virtualbox  --virtualbox-cpu-count "2" --virtualbox-memory "%VBOX_MEMORY%" --engine-insecure-registry 172.30.0.0/16 --virtualbox-boot2docker-url https://github.com/boot2docker/boot2docker/releases/download/v1.13.1/boot2docker.iso openshift`

I had an old copy of the init.bat file that did **NOT** use the docker-machine instead it used this command to directly startup a cluster up:

`C:\OC\ocp-install-demo-master>call oc cluster up --image=registry.access.redhat.com/openshift3/ose --version=v3.6`

Output:

Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for registry.access.redhat.com/openshift3/ose:v3.6 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ...
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ...
   Using 10.0.75.2 as the server IP
-- Starting OpenShift container ...
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
FAIL
   **Error: cannot access master readiness URL https://10.0.75.2:8443/healthz/ready**
   Details:
     No log available from "origin" container

   Caused By:
     Error: Get https://10.0.75.2:8443/healthz/ready: authenticationrequired
**(This is probably because of a TLS Certificate error because I can manuall browse to that url on chrome ,ignore the cert and I get a status of "OK" )**

C:\OC\ocp-install-demo-master>if 1 NEQ 0 (
echo.
 echo There was an issue starting OCP. If you feel the need, raise an issue and add error output:
 echo.
 echo    https://github.com/redhatdemocentral/ocp-install-demo/issues/new
 echo.
 GOTO :EOF
)

There was an issue starting OCP. If you feel the need, raise an issue and add error output:

   https://github.com/redhatdemocentral/ocp-install-demo/issues/new

**(I still try to login )**
C:\OC\ocp-install-demo-master>oc login -u system:admin
Server [https://localhost:8443]:
**error: dial tcp 127.0.0.1:8443: connectex: No connection could be made because the target machine actively refused it. - verify you have provided the correct host and port and that the server is currently running.**

Then I try to do run a oc cluster up (again) it tells me OpenShift is already running:

C:\OC\ocp-install-demo-master>oc cluster up
Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.5 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
**-- Checking for existing OpenShift container ... FAIL
   Error: OpenShift is already running**
   Solution:
     To start OpenShift again, stop the current cluster:
     $ oc cluster down

Any help will be greatly appreciated as I have spent numerous hours trying to get a local cluster running on my PC. 

**ALSO Please note that Iam behind a corporate Firewall. But I have full admin rights on my PC.** 

I see the ose docker image when i run docker images.( It is also up and running )

C:\OC\ocp-install-demo-master>docker images
REPOSITORY                                  TAG                 IMAGE ID            CREATED             SIZE
registry.access.redhat.com/openshift3/ose   v3.6                04571bc6f578        2 weeks ago         970MB

So looks like Iam close .... any help will be greatly appreciated.

Thanks a lot for your time it is much appreciated.

**Comment by sg1970** [comment]
@eschabell 

Out of curiosity I tried to access the Web Console using https://10.0.75.2:8443 (despite **init.bat** being unsuccessful ) and it worked after I accepted the TLS Cert Warning on my browser. I was also able to login as system and create a test project but there are no image streams available . I cannot create the image streams from command line as it wont let me login from CLI to execute the oc create commands.

Any help will be greatly appreciated to get me over the CLI issues.

Thanks!

-----------

-- **Login using Token Generated from Web GUI.**
C:\OC\ocp-install-demo-master>oc login https://10.0.75.2:8443 --token=a59Zzu6F9CnKGlQb6pwweP5yPborLu-s2yPCtg6FILOA
**error**: authenticationrequired

-- **Login Using default ID/passwd for system user.**
C:\OC\ocp-install-demo-master>oc login https://10.0.75.2:8443 -u system:admin
**error**: authenticationrequired

C:\OC\ocp-install-demo-master>oc create -n openshift -f https://raw.githubusercontent.com/redhat-developer/s2i-dotnetcore/master/dotnet_imagestreams.json
error: Missing or incomplete configuration info.  Please login or point to an existing, complete config file:

  1. Via the command-line flag --config
  2. Via the KUBECONFIG environment variable
  3. In your home directory as ~/.kube/config

To view or setup config directly use the 'config' command.

C:\OC\ocp-install-demo-master>

**Comment by eschabell** [comment]
@sg1970, Looks like you have a partial install and for some reason your streams can't be installed....   it just looks like the install was partially completed, most likely just a plain empty OCP without the product image streams. 

The login you need for CLI to install image streams should be what you are using:

$ oc login -u system:admin

Then on unix systems you have to evaluate the command to set environment up:

$ eval "$(docker-machine env openshift)"

For windows the init.bat does not have this, but not sure if it's needed? I think once the environment is set for windows, your stream installations might start working.

Can you verify that for me?

**Comment by sg1970** [comment]
@eschabell 

Not sure I understand what you are saying ... since the older version of int.bat file does **NOT** run the docker-machine create command to create the docker machine named "openshift"  ( like the current version of the bat file does. And Iam using the commands from the old init.bat file due to me not having Virtual-Box  as Iam using Hyper-V hypervisor on Windows 10 Pro). See the 3rd Post in this thread for details.

The Command I am referring to is:
`docker-machine create --driver  virtualbox --virtualbox-cpu-count "2" --virtualbox-memory "%VBOX_MEMORY%" --engine-insecure-registry 172.30.0.0/16 --virtualbox-boot2docker-url https://github.com/boot2docker/boot2docker/releases/download/v1.13.1/boot2docker.iso openshift`

I will be more than happy to try an equivalent command for the Hyper-V driver if you can send me the command.

**Comment by eschabell** [comment]
Let me check with a teammate on this.....

@sabre1041, any ideas how we could support this on windows outside of virtualbox? Possible in our demo use case?

**Comment by sg1970** [comment]
@eschabell thanks for the help! I edited my previous post to structure it better to avoid any confusions.

**Comment by sabre1041** [comment]
@eschabell @sg1970  hyper-v adds a few more steps that would be necessary.

First, as a prerequisite, a new virtual network must be created. Otherwise, the incorrect network is utilized for docker. More information on how to create the hyper-v virtual network can be found [here](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/connect-to-network).

Once the network has been defined, a similar `docker-machine create` command can be utilized. The only difference is to swap virtualbox with hyperv in the switches. An additional switch called `--hyperv-virtual-switch` can reference the name of the network defined previously. 

An example command to create the docker machine can be found below (All commands must be run with administrative privileges!):

```
docker-machine create --driver hyperv --hyperv-cpu-count "2" --hyperv-virtual-switch "External (Wireless)" --hyperv-memory "2048" --engine-insecure-registry 172.30.0.0/16 --hyperv-boot2docker-url https://github.com/boot2docker/boot2docker/releases/download/v1.13.1/boot2docker.iso openshift
```

Once the docker machine has been defined, load the newly created environment variables into the current environment. I ran into issues with powershell, so I tested with the Command Prompt:

```
@FOR /f "tokens=*" %i IN ('docker-machine env openshift') DO @%i
```

Finally call _oc cluster up_ as per usual

```
oc cluster up --image=registry.access.redhat.com/openshift3/ose --host-data-dir=/var/lib/boot2docker/ocp-data  --docker-machine=openshift --host-config-dir=/var/lib/boot2docker/ocp-config --use-existing-config=true --host-pv-dir=/var/lib/boot2docker/ocp-pv
```

**Comment by sg1970** [comment]
@sabre1041  thanks for the response unfortunately it did not work for me .. but it did go a lot further than before so this is progress !!.  Also despite the error I can connect to the openshift vm from the Hyper-V Manager gui and login as root. 

Can you post your Docker version and Windows Version info and which mode (Linux/Windows) are you running ?  Does the docker-machine create command complete successfully for you ? 

The error I get: **Error creating machine: Error detecting OS: OS type not recognized**

**Details:**

1. To get the VM Switch name I run  Get-VMSwitch from powershell.
2. I manually downloaded the boot2docker.iso file from the github URL in your post and host it on my local IIS Web Server to get around the proxy issue ( it just does not work behind a corporate http proxy  ... I get authntication error). 
3.  Docker tray is in Linux mode.
4. Run the docker-machine create command in your post ( with URL changed to point to  http://localhost/iso/boot2docker.iso )
5. Get the error :  Error creating machine: Error detecting OS: OS type not recognized
6. Repeat with Docker in Windows Container Mode" ( Right Click on Docker Tray icon --> Switch to Windows Container ). I get the same error. See below.

**Commands:**

`PS C:\Windows\system32> Get-VMSwitch`

Name      SwitchType NetAdapterInterfaceDescription
----      ---------- ------------------------------
nat       Internal                                 
DockerNAT Internal 

`c:\>docker-machine create --driver hyperv --hyperv-cpu-count "2" --hyperv-virtual-switch "nat" --hyperv-memory "2048" --engine-env HTTP_PROXY=http://sg1970:pswd@proxy.xyz.com:8080 --engine-env HTTPS_PROXY=https://sg1970:pswd@proxy.xyz.com:8080  --engine-insecure-registry 172.30.0.0/16 --hyperv-boot2docker-url http://localhost/iso/boot2docker.iso openshift`
Running pre-create checks...
Creating machine...
(openshift) Downloading C:\Users\sg1970\.docker\machine\cache\boot2docker.iso from http://localhost/iso/boot2docker.iso...
(openshift) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
(openshift) Creating SSH key...
(openshift) Creating VM...
(openshift) Using switch "nat"
(openshift) Creating VHD
(openshift) Starting VM...
(openshift) Waiting for host to start...
Waiting for machine to be running, this may take a few minutes...
Detecting operating system of created instance...
Waiting for SSH to be available...
Detecting the provisioner...
**Error creating machine: Error detecting OS: OS type not recognized**

Appreciate your help.

**Comment by sabre1041** [comment]
@sg1970 as described previously, you need to create an external virtual network, otherwise it will use the internal Docker network and fail as you are experiencing. When running the `docker-machine create` command, you will need to reference the name using the `--hyperv-virtual-switch` switch

**Comment by sg1970** [comment]
@sabre1041 Thanks for spotting that. I was able to exec the docker-machine create command and set my env using `@FOR /f "tokens=*" %i IN ('docker-machine env openshift') DO @%i`

But it does not start the cluster. I get the "authenticationrequired" error which I suspect is either a certificate issue or the proxy issue. ( Rant: Must say that docker/openshift tech stack is so  blissfully ignorant of the world that has to live behind a proxy but thats a topic for another day ... )


`c:\OC\ocp-install-demo-master>oc cluster up --image=registry.access.redhat.com/openshift3/ose --host-data-dir=/var/lib/boot2docker/ocp-data  --docker-machine=openshift --host-config-dir=/var/lib/boot2docker/ocp-config --use-existing-config=true --host-pv-dir=/var/lib/boot2docker/ocp-pv`

Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.5 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ...
   WARNING: Cannot verify Docker version
-- Checking for existing OpenShift container ... FAIL
   Error: unexpected error while checking OpenShift container state
   Caused By:
     **Error: An error occurred trying to connect: Get https://10.123.45.192:2376/containers/origin/json: authenticationrequired**

Infact I cannot even exec a simple `docker version` command successfully.

`c:\OC\ocp-install-demo-master>docker version`
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:48:20 2017
 OS/Arch:      windows/amd64
**error during connect: Get https://10.123.45.192:2376/v1.30/version: authenticationrequired**

Help !!!

**Comment by sg1970** [comment]
@sabre1041 

I re-created the docker-machine with these switches (and also using the github url for the boot2docker.iso) :   --engine-env HTTP_PROXY=http://usr:pwd@proxy.xyz.com:port  ( which downloaded the boot2docker.iso  from github and created the openshift vm succesfully .... therefore I think I have addressed the Proxy issue ). 

After this I went ahead and ran the oc cluster up command and now I get a different error message **: Cannot connect to the Docker daemon**

`c:\OC\ocp-install-demo-master>oc cluster up --image=registry.access.redhat.com/openshift3/ose --host-data-dir=/var/lib/boot2docker/ocp-data  --docker-machine=openshift --host-config-dir=/var/lib/boot2docker/ocp-config --use-existing-config=true --host-pv-dir=/var/lib/boot2docker/ocp-pv`
Starting OpenShift using registry.access.redhat.com/openshift3/**ose:v3.6.173.0.5 ...**
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ...
   WARNING: Cannot verify Docker version
-- Checking for existing OpenShift container ... FAIL
   **Error: unexpected error while checking OpenShift container state
   Caused By:
     Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?**

Onething I notice is that the openshift image which it is trying to run has a different tag than the one I see listed when I run docker images . 

c:\>docker images
REPOSITORY                                  TAG                 IMAGE ID            CREATED             SIZE
registry.access.redhat.com/openshift3/ose   **v3.6**                04571bc6f578        4 weeks ago         970MB

I will try to delete this image and run the cluster up command to see if it downloads the correct image again.

**Comment by sg1970** [comment]
@sabre1041 

I tried dropping the image , re-creating the docker-machine create for openshift and run the oc cluster up and I run into the same error.

`c:\OC\ocp-install-demo-master>docker version`
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:48:20 2017
 OS/Arch:      windows/amd64

Server:
 Version:      17.06.1-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:54:55 2017
 OS/Arch:      linux/amd64
 Experimental: true

`c:\>docker ps -a`
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

`c:\>docker rmi 04571bc6f578`
Untagged: registry.access.redhat.com/openshift3/ose:v3.6
Untagged: registry.access.redhat.com/openshift3/ose@sha256:b29e129d432bdedc32ff832f8c3f8e20decdde4577608ebe1790580bd7a8303f
Deleted: sha256:04571bc6f57814aa2b57e2eaedb9c7ee828ec8b16eb12d19c3c200dfa994c47a
Deleted: sha256:e95a140fb27a5cd65a9c607d846c6dab2e7622331297c221d5c5dd3a263337d9
Deleted: sha256:8bd7b79c41918075a0c20915ba97ad96aacd86e3525b1f543498cf46b5316be0
Deleted: sha256:1ce3b034f76a2aaa6324fed0cb802fe5a65fe5a9efbe6de98ed7cd9710cec17f
Deleted: sha256:f1f88d1c363a4390a107d572cc0f39c695848d078eda931c467fb88b48830ef2

`c:\OC\ocp-install-demo-master>docker images`
REPOSITORY                     TAG                 IMAGE ID            CREATED             SIZE
ubuntu                         latest              14f60031763d        7 weeks ago         120MB
webapi_dockersupport           dev                 640496b16054        2 months ago        305MB
eshop/webmvc                   dev                 7fb1813d3419        2 months ago        305MB
eshop/catalog.api              dev                 87aed5e592fb        2 months ago        305MB
eshop/ordering.api             dev                 563192d7c62d        2 months ago        305MB
eshop/webspa                   dev                 3976415863f9        2 months ago        305MB
eshop/marketing.api            dev                 678fe8ff8c90        2 months ago        305MB
eshop/identity.api             dev                 d882f5e97642        2 months ago        305MB
eshop/basket.api               dev                 93dc146c2a1a        2 months ago        305MB
eshop/locations.api            dev                 5c7c22bf8f66        2 months ago        305MB
eshop/webstatus                dev                 150b378d6eea        2 months ago        305MB
mongo                          latest              71c101e16e61        2 months ago        358MB
microsoft/aspnetcore           1.1                 7a07a406c6af        2 months ago        305MB
microsoft/aspnetcore           1.1.2               7a07a406c6af        2 months ago        305MB
rabbitmq                       latest              79e8fb6325bc        2 months ago        124MB
redis                          latest              83744227b191        3 months ago        98.9MB
microsoft/mssql-server-linux   latest              5985832855cf        4 months ago        1.38GB
d4w/nsenter                    latest              9e4f13a0901e        11 months ago       83.8kB

`c:\OC\ocp-install-demo-master>docker-machine env openshift`
SET DOCKER_TLS_VERIFY=1
SET DOCKER_HOST=tcp://10.123.45.192:2376
SET DOCKER_CERT_PATH=C:\Users\sg1970\.docker\machine\machines\openshift
SET DOCKER_MACHINE_NAME=openshift
SET COMPOSE_CONVERT_WINDOWS_PATHS=true
REM Run this command to configure your shell:
REM     @FOR /f "tokens=*" %i IN ('docker-machine env openshift') DO @%i

`c:\OC\ocp-install-demo-master> @FOR /f "tokens=*" %i IN ('docker-machine env openshift') DO @%i`

`c:\OC\ocp-install-demo-master>oc cluster up --image=registry.access.redhat.com/openshift3/ose --host-data-dir=/var/lib/boot2docker/ocp-data  --docker-machine=openshift --host-config-dir=/var/lib/boot2docker/ocp-config --use-existing-config=true --host-pv-dir=/var/lib/boot2docker/ocp-pv`
Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.5 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ...
   WARNING: Cannot verify Docker version
-- Checking for existing OpenShift container ... FAIL
   Error: unexpected error while checking OpenShift container state
   Caused By:
     Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?


--- These 2 errors (below) are due to the SET commands that we have to run above. They work fine on a different command window where those SET commands have not been executed.

`c:\OC\ocp-install-demo-master>docker images`
**error during connect: Get https://10.123.45.192:2376/v1.30/images/json: x509: cannot validate certificate for 10.123.45.192 because it doesn't contain any IP SANs**

`c:\OC\ocp-install-demo-master>docker version`
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:48:20 2017
 OS/Arch:      windows/amd64
**error during connect: Get https://10.123.45.192:2376/v1.30/version: x509: cannot validate certificate for 10.123.45.192 because it doesn't contain any IP SANs**

**Comment by sg1970** [comment]
@sabre1041 
just out of curiosity I executed just the "oc cluster up" command and it went further though still incomplete with a different error.

**Details**:

**c:\OC\ocp-install-demo-master>oc cluster up**
Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.5 ...
Pulling image registry.access.redhat.com/openshift3/ose:v3.6.173.0.5
Pulled 1/4 layers, 28% complete
Pulled 2/4 layers, 71% complete
Pulled 3/4 layers, 87% complete
Pulled 4/4 layers, 100% complete
Extracting
Image pull complete
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for registry.access.redhat.com/openshift3/ose:v3.6.173.0.5 image ...
   Pulling image registry.access.redhat.com/openshift3/ose:v3.6.173.0.5
   Pulled 1/4 layers, 28% complete
   Pulled 2/4 layers, 71% complete
   Pulled 3/4 layers, 87% complete
   Pulled 4/4 layers, 100% complete
   Extracting
   Image pull complete
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ...
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ...
   Using 10.0.75.2 as the server IP
-- Starting OpenShift container ...
   Creating initial OpenShift configuration
   Starting OpenShift using container 'origin'
   Waiting for API server to start listening
**FAIL
   Error: cannot access master readiness URL https://10.0.75.2:8443/healthz/ready
   Details:
     No log available from "origin" container

   Caused By:
     Error: Get https://10.0.75.2:8443/healthz/ready: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)**

`c:\OC\ocp-install-demo-master>docker images`
REPOSITORY                                  TAG                 IMAGE ID            CREATED             SIZE
registry.access.redhat.com/openshift3/ose   **v3.6.173.0.5**        a00c5ef1fe3d        2 weeks ago         970MB
ubuntu                                      latest              14f60031763d        7 weeks ago         120MB
webapi_dockersupport                        dev                 640496b16054        2 months ago        305MB
eshop/webmvc                                dev                 7fb1813d3419        2 months ago        305MB
eshop/ordering.api                          dev                 563192d7c62d        2 months ago        305MB
eshop/catalog.api                           dev                 87aed5e592fb        2 months ago        305MB
eshop/marketing.api                         dev                 678fe8ff8c90        2 months ago        305MB
eshop/webspa                                dev                 3976415863f9        2 months ago        305MB
eshop/basket.api                            dev                 93dc146c2a1a        2 months ago        305MB
eshop/identity.api                          dev                 d882f5e97642        2 months ago        305MB
eshop/locations.api                         dev                 5c7c22bf8f66        2 months ago        305MB
eshop/webstatus                             dev                 150b378d6eea        2 months ago        305MB
mongo                                       latest              71c101e16e61        2 months ago        358MB
microsoft/aspnetcore                        1.1                 7a07a406c6af        2 months ago        305MB
microsoft/aspnetcore                        1.1.2               7a07a406c6af        2 months ago        305MB
rabbitmq                                    latest              79e8fb6325bc        2 months ago        124MB
redis                                       latest              83744227b191        3 months ago        98.9MB
microsoft/mssql-server-linux                latest              5985832855cf        4 months ago        1.38GB
d4w/nsenter                                 latest              9e4f13a0901e        11 months ago       83.8kB

c:\OC\ocp-install-demo-master>

**Comment by eschabell** [comment]
One problem I see is that you are using oc client version v3.6.173.0.5 and there is a new one that you need: v3.6.173.0.21

**Comment by sg1970** [comment]
@eschabell  @sabre1041 

No luck. I got access to that version (v3.6.173.0.21) of oc.exe and ran the oc cluster up command. I got the exact same error(see err msg below). 

**Question**: Has this ever worked in a corporate environment thats behind a Proxy?.  I suspect not but please clarify.

---- ERROR ----

Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.21 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ...
   WARNING: Cannot verify Docker version
-- Checking for existing OpenShift container ... FAIL
   Error: unexpected error while checking OpenShift container state
   Caused By:
     Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?

**Comment by sg1970** [comment]
@eschabell @sabre1041 

Have we hit a bug? any ideas?

**Comment by eschabell** [comment]
@sg1970 last error is no docker being found on the host trying to install Openshift image on... I can't replicate as don't have your platform, but need to find out why it can't find a docker to work with?

**Comment by sg1970** [comment]
@eschabell  

1. I notice is that the docker version was **1.13.1** for the openshift machine that was created using docker-machine create command(see attached image below). I downloaded the boot2docker.iso for version 17.06 and re-created the openshift machine and tried  this:

`oc cluster up --image=registry.access.redhat.com/openshift3/ose ......` 

 but unfortunately I still get the same error:

**Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.21 ...
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ...
WARNING: Cannot verify Docker version
-- Checking for existing OpenShift container ... FAIL
Error: unexpected error while checking OpenShift container state
Caused By:
Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?**

2. I also notice that the above warning for docker version does **not** occur when I use **just** the `oc cluster up ` command.  I believe this runs on the default mobylinux machine? Is there any difference between the 2 machines ?

3.  Has this ever been tested on a strict corporate env ? Just wanted to know if it ever worked for anyone else.

4. Is it possible for you to have this escalated within Redhat ?

Thanks for your time.

------

![openshift_vm](https://user-images.githubusercontent.com/19354875/30597408-e5569e20-9d24-11e7-8a42-ba1eb1e5c474.PNG)

**Comment by eschabell** [comment]
@sg1970 please understand that 'oc cluster up' is never intended for corporate installations, it's a demo environment tool. This project is an extension of that trying to provide a laptop experience for as far as that works within a non-vm environment.

We try to assist your venturing off the beaten path, but have very little possibility to test this ourselves (at least I don't).

Btw, the image created with docker 1.13.1 is the supported OCP internal docker engine version, which is what I pin this installation to. If you deviate from that, you are again leaving our path and venturing in to the unknown.

**Comment by sg1970** [comment]
@eschabell 

I was just trying to compare the behavior between `oc cluster up` and` oc cluster up --image=registry.access.redhat.com/openshift3/ose` ... to try and get some clues . I don't have any preference for which method we use as long as it works.  Same for docker version. 

While my PC is a corporate PC it is still meant for a demo/evaluation usecase. Nothing even remotely Production grade is intended.  

Once again thanks for your time and please let me know if we are looking at a dead end here.

**Comment by eschabell** [comment]
@sg1970, running a workshop this week using this and lucky enough to have someone in the class happy to test this out. He came up with this, can you try it please?

His adaption for init.bat:

_call docker-machine create --driver hyperv --hyperv-cpu-count "2" --hyperv-memory "10240" --hyperv-disk-size "10240" --engine-insecure-registry 172.30.0.0/16 --hyperv-boot2docker-url https://github.com/boot2docker/boot2docker/releases/download/v1.13.1/boot2docker.iso --hyperv-virtual-switch="ext-network" openshift_

"It is important to remark that you must create a virtual switch in your pc in advance."

Let us know?

**Comment by sg1970** [comment]
@eschabell 

That command just hangs at Waiting for host to start... and although I kinda got around it but the end result is it did not make any difference for oc cluster up. The Details below.

```
c:\>docker-machine create --driver hyperv --hyperv-cpu-count "2" --hyperv-memory "10240" --hyperv-disk-size "10240" --engine-insecure-registry 172.30.0.0/16 --hyperv-boot2docker-url http://localhost/boot2docker.iso --hype
rv-virtual-switch "Ext_Ethernet_Switch_1" openshift
```
**Output:**

> Running pre-create checks...
> Creating machine...
> (openshift) Downloading C:\Users\rsr1rxd\.docker\machine\cache\boot2docker.iso from http://localhost/boot2docker.iso...
> (openshift) 0%....10%....20%....30%....40%....50%....60%....70%....80%....90%....100%
> (openshift) Creating SSH key...
> (openshift) Creating VM...
> (openshift) Using switch "Ext_Ethernet_Switch_1"
> (openshift) Creating VHD
> (openshift) Starting VM...
> (openshift) Waiting for host to start...
> **_(Hangs here)_**

So I cancelled it and compared it to the previous docker-machine create command and noticed that it differed only in the Memory size and explicitly specified disk-size.  So I deleted the VM and recreated it **without** the  `--hyperv-disk-size ` option and it created the openshift VM which I can verify thru Hyper-V manager GUI. I can also log into it as root and verify docker version = 1.13.1.  

I then manually edited the disk size ( using Hyper-V manager GUI)  to be 20GB (it wouldnt let me size it any less than that ) and started the openshift VM succesfully. I then ran the oc cluster up command but alas it throws the same error as always ( Cannot connect to the Docker daemon. Is the docker daemon running on this host? )

`c:\OC\ocp-install-demo-master>oc cluster up --image=registry.access.redhat.com/openshift3/ose --host-data-dir=/var/lib/boot2docker/ocp-data --docker-machine=openshift --host-config-dir=/var/lib/boot2docker/ocp-config --use-existing-config=true --host-pv-dir=/var/lib/boot2docker/ocp-pv`

**Output**:

> 
> Starting OpenShift using registry.access.redhat.com/openshift3/ose:v3.6.173.0.21 ...
> -- Checking OpenShift client ... OK
> -- Checking Docker client ... OK
> -- Checking Docker version ...
>    WARNING: Cannot verify Docker version
> -- Checking for existing OpenShift container ... FAIL
>    Error: unexpected error while checking OpenShift container state
>    Caused By:
>      Error: Cannot connect to the Docker daemon. Is the docker daemon running on this host?

c:\OC\ocp-install-demo-master>

Is there a log file or some debug trace that we can look at somewhere to see whats causing it to error ?

**Comment by sg1970** [comment]
@eschabell 

I tried the following with no luck (Still get the exact same error message)

1.  Temporarily disabling the Windows Firewall and the Symantec Antivirus. 
2.  Tried with TLS disabled (SET DOCKER_TLS_VERIFY=0) and pointing to port 2375 ( using SET DOCKER_HOST=tcp://<openshift_VM_ip_address>:2375 )

But it does work beautifully on my home PC. I'am out of ideas at this point. 

Are there any ways to debug this issue?
